### PR TITLE
Fix Queries of Unsupported types appearing on Visualize & Vis Builder Search Bar

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

Fix Queries of Unsupported types appearing on Visualize Search Bar.

https://github.com/user-attachments/assets/f9bdfad0-957f-4da0-bd20-ca0998a99d06

## Testing the changes

https://github.com/user-attachments/assets/df6ac254-976b-41cc-8195-a4c251265164

## Changelog

- skip
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ashwin-pc — APPROVED** [review]
Looks good. Just one comment, but not a blocker for this issue. It can be a fast follow

File: src/plugins/data/public/query/query_string/query_string_manager.ts [hunk 1]
```diff
@@ -194,6 +231,29 @@ export class QueryStringManager {
...
+      console.log('Application Not available.');
+    }
+
+    return appId;
+  };
 }
 
+const showWarning = debounce((notifications, { title, text }) => {
+  notifications.toasts.addWarning({ title, text });
+}, 500);
```
Comment (ashwin-pc) [inline src/plugins/data/public/query/query_string/query_string_manager.ts:253]:
```suggestion
const showWarning = (notifications, { title, text }) => {
  notifications.toasts.addWarning({ title, text, id: 'id-to-prevent-rerender' });
};
```

**Review by ananzh — APPROVED** [review]

File: src/plugins/data/public/query/query_string/query_string_manager.ts [hunk 2]
```diff
@@ -194,6 +231,29 @@ export class QueryStringManager {
       this.uiSettings.get(UI_SETTINGS.SEARCH_QUERY_LANGUAGE)
     );
   }
+
+  private getCurrentAppId = () => {
+    let appId;
+    try {
+      const application = getApplication();
+      if (application) {
+        application.currentAppId$.subscribe((val) => (appId = val)).unsubscribe();
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply src/plugins/data/public/query/query_string/query_string_manager.ts:240]: Actually the getValue() is supported only for `BehaviorSubject<string | undefined>` but the currentAppId$ is a `Observable<string | undefined>` hence this method is not supported

File: src/plugins/visualize/public/application/utils/utils.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ export const addBadgeToAppChrome = (chrome: ChromeStart) => {
   });
 };
 
-export const getDefaultQuery = ({ data }: VisualizeServices) => {
+export const getDefaultQuery = ({ data, toastNotifications, uiSettings }: VisualizeServices) => {
```
[the remark this answers is not in the document]
Comment (LDrago27) [reply src/plugins/visualize/public/application/utils/utils.ts:63]: Thanks will remove the unused dependencies.

File: src/plugins/data/public/query/query_string/query_string_manager.ts (hunk 1 above)
Comment (LDrago27) [reply src/plugins/data/public/query/query_string/query_string_manager.ts]: Thanks fixed it .

**Review by ananzh — APPROVED** [review]

**Review by virajsanghvi — APPROVED** [review]
approving assuming ci passes

**Pull request merged**